src/jit/simdcodegenxarch.cpp

```diff
@@ -619,8 +619,8 @@ CodeGen::genSIMDIntrinsicInit(GenTreeSIMD* simdNode)
             // data section using movss/sd.  Similarly if op1 is a memory op we
             // would have loaded it using movss/sd.  Movss/sd when loading a xmm reg
             // from memory would zero-out upper bits. In these cases we can
-            // avoid explicitly zero'ing out targetReg.
-            bool zeroInitRequired = !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
+            // avoid explicitly zero'ing out targetReg if targetReg and op1Reg are the same.
+            bool zeroInitRequired = (targetReg != op1Reg) || !(op1->IsCnsFltOrDbl() || op1->isMemoryOp());
```

Comment:
That will miss the
if (compiler->getSIMDInstructionSet() == InstructionSet_AVX)
logic from genSIMDScalarMove.